### PR TITLE
Hotfix: sanitize url param

### PR DESCRIPTION
**Pull request by ravinderk**
Resolves https://github.com/impress-org/givewp/issues/6473

## Description

URL param should be sanitized before nursing in business logic. I updated logic to sanitize 'class' URL param.

## Testing Instructions

You should able to export donations.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: includes/admin/tools/export/export-actions.php [hunk 1]
```diff
@@ -37,18 +38,20 @@ function give_process_batch_export_form() {
 
 	require_once GIVE_PLUGIN_DIR . 'includes/admin/tools/export/class-batch-export.php';
 
-	/**
+    $classname = give_clean($_REQUEST['class']);
```
Comment (JasonTheAdams) [inline includes/admin/tools/export/export-actions.php:41]: @ravinderk I'm afraid this isn't enough. It still means that the request can initialize **any** class. That's way too open. This should have a whitelist of classes that its able to construct.

Comment (ravinderk) [reply includes/admin/tools/export/export-actions.php:41]: @JasonTheAdams good catch. I will implement it

Comment (ravinderk) [reply includes/admin/tools/export/export-actions.php:41]: @JasonTheAdams We are using this feature in another add-on too:
- Gift Aid
- Recurring Donation
- Peer To Peer

We are not aware of classes upfront, so we can not whitelist all classes. What do you think about the validation class by `instanceof \Give_Batch_Export`?

Comment (JasonTheAdams) [reply includes/admin/tools/export/export-actions.php:41]: Yeah! That works! That's a good constraint. 👍

Comment (ravinderk) [reply includes/admin/tools/export/export-actions.php:44]: fixed: https://github.com/impress-org/givewp/compare/427d62bb9327...42834be8b970

Comment (ravinderk) [reply includes/admin/tools/export/export-actions.php:39]: fixed: https://github.com/impress-org/givewp/compare/42834be8b970...f44b25e891d8

**Review by JasonTheAdams — APPROVED** [review]
This looks secure to me, @ravinderk. Please have QA test this out to make sure we haven't broken anything.

**Review by Benunc — APPROVED** [review]
Rick says this works.

**Pull request merged**